File: src/main/java/com/tobe/healthy/common/KotlinCustomPaging.java
package com.tobe.healthy.common;

import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.tobe.healthy.common.NotificationSenderInfo.SenderInfo;
import com.tobe.healthy.notification.presentation.dto.out.NotificationRedDotStatusResult;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class KotlinCustomPaging<T> {

	private List<T> content;

	private int pageNumber;

	private int pageSize;

	private int totalPages;

	private long totalElements;

	@JsonProperty("isLast")
	private boolean isLast;

	@JsonInclude(Include.NON_EMPTY)
	@Builder.Default
	private List<NotificationRedDotStatusResult> redDotStatus = new ArrayList<>();

	@JsonInclude(Include.NON_NULL)
	private SenderInfo sender;

	public KotlinCustomPaging(
		List<T> content,
		int pageNumber,
		int pageSize,
		int totalPages,
		long totalElements,
		boolean isLast
	) {
		this.content = content;
		this.pageNumber = pageNumber;
		this.pageSize = pageSize;
		this.totalPages = totalPages;
		this.totalElements = totalElements;
		this.isLast = isLast;
		this.redDotStatus = new ArrayList<>();
		this.sender = null;
	}
}

File: src/main/kotlin/com/tobe/healthy/common/KotlinCustomPaging.kt
package com.tobe.healthy.common

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_EMPTY
import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
import com.tobe.healthy.common.NotificationSenderInfo.SenderInfo
import com.tobe.healthy.notification.domain.dto.out.NotificationRedDotStatusResult

data class KotlinCustomPaging<T>(
    var content: List<T>,
    val pageNumber: Int,
    val pageSize: Int,
    val totalPages: Int,
    val totalElements: Long,
    val isLast: Boolean,
    @JsonInclude(NON_EMPTY)
    val redDotStatus: List<NotificationRedDotStatusResult>? = mutableListOf(),
    @JsonInclude(NON_NULL)
    val sender: SenderInfo? = null,
)
